scipy/spatial/setlist.pxd

```diff
@@ -47,6 +47,12 @@ cdef inline int init(setlist_t *setlist, size_t n, size_t size_guess) except -1:
         setlist.sizes[j] = 0
         setlist.alloc_sizes[j] = size_guess
         setlist.sets[j] = <int*>libc.stdlib.malloc(sizeof(int) * size_guess)
+        if setlist.sets[j] == NULL:
+            for i in range(j):
+                libc.stdlib.free(<void*>setlist.sets[i])
+            libc.stdlib.free(<void*>setlist.sets)
+            libc.stdlib.free(<void*>setlist.sizes)
```

Comment:
Looks like I forgot to click on the "add comment" button for one of my comments in my last review.  We also need to free `setlist.alloc_sizes` here.
```suggestion
            libc.stdlib.free(<void*>setlist.sizes)
            libc.stdlib.free(<void*>setlist.alloc_sizes)
```